sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -988,8 +988,20 @@ case class CatalystToExternalMap private(
   override def children: Seq[Expression] =
     keyLambdaFunction :: valueLambdaFunction :: inputData :: Nil
 
-  override def eval(input: InternalRow): Any =
-    throw new UnsupportedOperationException("Only code-generated evaluation is supported")
+  private lazy val toScalaValue: Any => Any = {
+    assert(inputData.dataType.isInstanceOf[MapType])
+    val mapType = inputData.dataType.asInstanceOf[MapType]
+    CatalystTypeConverters.createToScalaConverter(mapType)
```

Comment:
hmm, don't we need to support `collClass`?

Reply:
yea, you're right. I'll update soon.

Reply:
We already have a helper function for `Class[_] -> DataType`? It seems the most similar one is `HiveInspectors.javaTypeToDataType`?

Reply:
ok, I'll consider again.